### ISSUE TITLE
postBuildCommands doesn't run again even if failed

### ISSUE THREAD

**Issue by atilaneves** [dlang/dub#1047]
Let's say my `postBuildCommands` is `"./foo > foo.txt"`. If that command fails for any reason, running `dub build` again won't do anything. I think the solution here is to write to a file in the `.dub` directory with the return code of the last `postBuildCommands` and run them again even if the main target is up to date.

**Comment by andre2007** [comment]
I assume adding this feature would add more trouble than it solves.
Dub can be called with --force which forces a rebuild and therefore the postBuildCommands are triggered again.
Also in the meantime there are pre/post run command hooks added, which are executed on every run. This might also solve some use cases.

**Comment by atilaneves** [comment]
Neither `--force` or `postRunCommands` do what the use would want, which is to run commands only after a succesful build as a dependency. Why would I want to go through the pain of building every source files again if they haven't changed just to  run a command?

When people complain that dub isn't a good build system, this is what they mean. It's not a build system at all when it can't do what make has been doing for decades.

**Comment by andre2007** [comment]
I fully understand your complains. The solution is more complex than just a single file.
You could have different configurations in your dub.json. for each you need this file. Also postBuildCommands could be mentioned several times within a configurations (for different os).
Also In case the postBuildCommands changes, this must be detected. Until now I haven't found a good solution for this issue, but your requirement makes sense.

**Comment by s-ludwig** [comment]
I think a single dummy file output to `.dub/build/.../` after successful completion and adding that to the list of dependency files in `isUpToDate` should be sufficient to fix this.
